<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.27">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Quality Control of Sequencing Data – EMPLANT course</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-b754b9fd3853d7e56faef54c29c5d52b.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-7ff69c1ec09585e6d40427abec3a7d0f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/intersection-observer-polyfill-1.0.0/intersection-observer.js"></script>
<script src="site_libs/quarto-contrib/scrollama-3.2.0/scrollama.min.js"></script>
<script src="site_libs/quarto-contrib/closeread-0.1.0/closeread.js"></script>
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<meta cr-debug-mode="false">
<meta cr-remove-header-space="false">


<link rel="stylesheet" href="site_libs/quarto-contrib/quarto-project/qmd-lab/closeread/closeread.css">
</head>

<body class="nav-sidebar docked nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">EMPLANT course</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./data_management.html" aria-current="page"> 
<span class="menu-text">Reproducibility</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./qc.html"> 
<span class="menu-text">Quality control</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" role="link" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/amrei-bp/emplant_bioinformatics_course">
            Source Code
            </a>
          </li>
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/amrei-bp/emplant_bioinformatics_course/issues">
            Report a Bug
            </a>
          </li>
      </ul>
    </div>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./qc.html">Quality control</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./figures/logo/SLUBI_logo.png" alt="" class="sidebar-logo light-content py-0 d-lg-inline d-none">
      <img src="./figures/logo/SLUBI_logo.png" alt="" class="sidebar-logo dark-content py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./data_management.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Reproducibility</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./qc.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Quality control</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content column-page-right" id="quarto-document-content">
      <!-- user-provided yaml closeread style options -->
      <style>
        :root {
          --cr-section-background-color: #ffffff;
        }
      </style>
    

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Quality Control of Sequencing Data</h1>
</div>



<div class="quarto-title-meta column-page-right">

    
  
    
  </div>
  


</header>


<p><br> By the end of this module, you should be able to:</p>
<ol type="1">
<li>Explain why QC of sequencing data is necessary.</li>
<li>Interpret FastQC reports and their warnings.</li>
<li>Decide what action (if any) to take based on QC results.</li>
</ol>
<section id="from-plant-tissue-to-sequencing-reads" class="level2">
<h2 class="anchored" data-anchor-id="from-plant-tissue-to-sequencing-reads">From plant tissue to sequencing reads</h2>
<p>Before we talk about quality control, it is important to understand how sequencing data is produced. Many of the problems we detect during QC originate early in the sequencing workflow.</p>
<p>At a high level, the process looks like this:</p>
<blockquote class="blockquote">
<p>Plant tissue → DNA extraction → Library preparation → Sequencing → FASTQ files</p>
</blockquote>
<p>Each step can introduce technical effects that later appear in quality control reports. Let’s have a closer look:</p>
<div class="cr-section cr-column-screen overlay-center">
<div class="narrative-col">
<div class="trigger new-trigger" data-focus-on="cr-sticky_sampling">
<div class="narrative">
<p>We start with <strong>plant tissue</strong> and extract the contained DNA.</p>
</div>
</div>
<div class="trigger new-trigger" data-focus-on="cr-sticky_prep">
<div class="narrative">
<p>Next, sequencing machines do not read whole chromosomes. Instead, DNA is broken into <strong>many small fragments</strong>, the so called <strong>DNA library</strong>.</p>
</div>
</div>
<div class="trigger new-trigger" data-focus-on="cr-sticky_seq">
<div class="narrative">
<p>The <strong>DNA library</strong> (with adapters added), is then run on a <strong>sequencing machine</strong>, which calls the bases of each fragment.</p>
</div>
</div>
<div class="trigger">
<div class="narrative">
<p>The final output are usually <strong>FASTQ files</strong>, which contain raw reads and quality scores.</p>
</div>
</div>
</div>
<div class="sticky-col">
<div class="sticky-col-stack">
<div id="cr-sticky_sampling" class="sticky">
<p><img src="figures/qc/sampling.png" class="img-fluid"></p>
</div>
<div id="cr-sticky_prep" class="sticky">
<p><img src="figures/qc/library_prep.png" class="img-fluid"></p>
</div>
<div id="cr-sticky_seq" class="sticky">
<p><img src="figures/qc/sequencing.png" class="img-fluid"></p>
</div>
</div>
</div>
</div>
</section>
<section id="why-do-we-need-quality-control" class="level2">
<h2 class="anchored" data-anchor-id="why-do-we-need-quality-control">Why do we need quality control?</h2>
<p>When we work with sequencing data, we usually want to answer a <strong>biological question</strong>.</p>
<p>In plant breeding, these could for example be:</p>
<ul>
<li>Which genetic variants are associated with a certain trait?</li>
<li>Which alleles are present in a breeding line?</li>
<li>How similar are two plant populations?</li>
<li>Which genes are expressed under certain conditions?</li>
</ul>
<p>All of these questions rely on the assumption that <strong>the sequencing data reflects the real biology of the plant.</strong></p>
<p>Quality control (QC) is the step where we check whether this assumption is reasonable.</p>
<p>As such, QC is not about making the data “look nice”. It is about making sure that <strong>technical problems do not lead to wrong biological conclusions</strong>.</p>
</section>
<section id="what-can-go-wrong-if-data-quality-is-poor" class="level2">
<h2 class="anchored" data-anchor-id="what-can-go-wrong-if-data-quality-is-poor">What can go wrong if data quality is poor?</h2>
<p>Sequencing errors do not stay “technical”. They can directly affect how we interpret the biology:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 45%">
<col style="width: 54%">
</colgroup>
<thead>
<tr class="header">
<th>Technical issue</th>
<th>Biological consequence</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Low base quality</td>
<td>Identification of false SNPs</td>
</tr>
<tr class="even">
<td>Adapter contamination</td>
<td>Detection of artificial variants</td>
</tr>
<tr class="odd">
<td>GC bias</td>
<td>Skewed allele frequency estimates</td>
</tr>
<tr class="even">
<td>High duplication</td>
<td>Overconfidence in genotypes</td>
</tr>
</tbody>
</table>
<p><br> In a breeding context, this can mean that we might <strong>choose the wrong markers</strong>, <strong>miss true marker-trait associations</strong> or <strong>misjudgement or genetiv diversity</strong>.</p>
<p>Quality control helps us detect these problems <em>before</em> we trust the results.</p>
</section>
<section id="where-do-sequencing-errors-come-from" class="level2">
<h2 class="anchored" data-anchor-id="where-do-sequencing-errors-come-from">Where do sequencing errors come from?</h2>
<p>Sequencing errors can be introduced at several steps, long before we ever see a FASTQ file.</p>
<section id="step-1-dna-extraction" class="level3">
<h3 class="anchored" data-anchor-id="step-1-dna-extraction">Step 1: DNA extraction</h3>
<p><img src="figures/qc/sampling.png" class="class img-fluid" style="width:40.0%"></p>
<p>Plant material can be challenging to work with. Different issuea can affect the quality of the reads that will be sequenced:</p>
<ul>
<li>Compounds such as polyphenols and polysaccharides can interfere with enzymes or make the DNA sticky.</li>
<li>DNA quality can vary strongly between samples, depending on the extraction method, tissue and extractor.</li>
</ul>
</section>
<section id="step-2-library-preparation" class="level3">
<h3 class="anchored" data-anchor-id="step-2-library-preparation">Step 2: Library preparation</h3>
<p><img src="figures/qc/library_prep.png" class="class img-fluid" style="width:40.0%"></p>
<p>During library preparation, the DNA is sheared into fragments, adapters are ligated to these DNA fragments and the DNA is often amplified using PCR.</p>
<p>Sometimes, the DNA already degrades before the library preparation. That can lead to <strong>uneven sequencing coverage</strong> in the sequenced reads. The <strong>PCR can introduce a bias</strong>, where some fragments are amplified more than others.</p>
<p>All these effects are <em>technical</em>, but they <strong>shape the data we later analyse</strong>.</p>
</section>
<section id="sequencing" class="level3">
<h3 class="anchored" data-anchor-id="sequencing">Sequencing</h3>
<p><img src="figures/qc/sequencing.png" class="class img-fluid" style="width:40.0%"></p>
<p>Sequencing run itself can introduce errors. Base calling becomes less accurate toward the end of reads and errors accumulate across sequencing cycles. Different lanes or runs can have different overall quality.</p>
</section>
</section>
<section id="why-plant-genomes-are-special" class="level2">
<h2 class="anchored" data-anchor-id="why-plant-genomes-are-special">Why plant genomes are special</h2>
<p>Many QC tools are <strong>generic</strong>. They do not “know” that your data comes from plants.</p>
<p>However, plant genomes often have features that affect quality:</p>
<ul>
<li>Large genome sizes</li>
<li>High repeat content</li>
<li>Polyploidy</li>
<li>High heterozygosity</li>
<li>Variable GC content between species</li>
</ul>
<p>As a result, the GC content plots may look unusual even for good data. High duplication levels are not always a technical problem. So, some warnings are expected and acceptable.</p>
<p>This means that <strong>QC results must always be interpreted in a biological context</strong>.</p>
<p>Quality control tools can show you <em>what</em> looks unusual — you need biological knowledge to decide <em>why</em>.</p>
</section>
<section id="what-is-fastqc" class="level2">
<h2 class="anchored" data-anchor-id="what-is-fastqc">What is FastQC?</h2>
<p>After sequencing, the first files we usually receive are FASTQ files. These files contain millions of short sequencing reads together with a quality score for every base.</p>
<p><strong>Looking directly at raw FASTQ files is not practical.</strong></p>
<p>FastQC is a tool for quality control on raw sequence data coming from high throughput sequencing. It takes your raw data and performs a series of standardized tests, and produces an <strong>easy-to-read overview report</strong> with figures and tables for each sample.</p>
<p>FastQC provides information about your samples, it does not know what experiment you performed, or what you plan to do with your data afterwards. Thus, the <strong>interpretation and decisions are up to you.</strong> After checking the quality you might want to use other tools to correct errors, remove low-quality data, etc.</p>
</section>
<section id="interpreting-fastqc-flags" class="level2">
<h2 class="anchored" data-anchor-id="interpreting-fastqc-flags">Interpreting FastQC flags</h2>
<p>Each FastQC module ends with a simple label: <strong>pass</strong>, <strong>warning</strong>, or <strong>fail</strong>. These labels are often misunderstood. A fail does not mean the data is unusable. Alternatively, a pass does not mean the data is perfect.</p>
<p>The flags are based on generic thresholds that work reasonably well across many datasets, but they are not tailored to specific organisms or analyses. In plant breeding projects, it is common to see warnings or failures even in perfectly usable datasets.</p>
<p>FastQC flags should therefore be treated as <strong>signals to investigate</strong>, not as automatic decisions.</p>
</section>
<section id="a-fastqc-report" class="level2">
<h2 class="anchored" data-anchor-id="a-fastqc-report">A FastQC report</h2>
<p>The following is adapted from the excellent <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">FastQC tutorial at Babraham Bioinformatics</a>.</p>
<section id="basic-statistics" class="level3">
<h3 class="anchored" data-anchor-id="basic-statistics"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/1%20Basic%20Statistics.html">Basic statistics</a></h3>
<p><img src="figures/qc/bsic_stats_qc.png" class="class img-fluid" style="width:50.0%"></p>
<p>Simple summary statistics for the file that was analyzed. Good to check what sample was used, and to get initial stats on the sample. This section will never raise warnings.</p>
</section>
<section id="per-base-sequence-quality" class="level3">
<h3 class="anchored" data-anchor-id="per-base-sequence-quality"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/2%20Per%20Base%20Sequence%20Quality.html">Per base sequence quality</a></h3>
<p><img src="figures/qc/per_base_qc.png" class="class img-fluid" style="width:50.0%"></p>
<p>This plot shows the <strong>quality score at each position in the read</strong>. The higher the score, the better the base call.</p>
<p>This plot answers a simple question: <em>How confident are we in the base calls along the read?</em></p>
<p>In most sequencing datasets, quality is highest at the beginning of the read and decreases toward the end. This pattern is normal and expected, a modest drop in quality at the end of reads is usually not a serious problem.</p>
<p>However, the sample can be problematic when:</p>
<ul>
<li>Quality drops very sharply.</li>
<li>Large parts of the read have very low scores.</li>
<li>Quality varies strongly between samples.</li>
</ul>
</section>
<section id="per-sequence-quality-scores" class="level3">
<h3 class="anchored" data-anchor-id="per-sequence-quality-scores"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/3%20Per%20Sequence%20Quality%20Scores.html">Per Sequence Quality Scores</a></h3>
<p><img src="figures/qc/sequence_quality.png" class="class img-fluid" style="width:50.0%"></p>
<p>While the previous plot looks at quality <strong>per position</strong>, this plot looks at quality <strong>per read</strong>.</p>
<p>It allows you to see if a subset of your sequences have low quality values, or if the quality is homogeneous across the reads of your sample.</p>
<p>The low quality reads should obviously only represent a small percentage of the total sequences. If a significant proportion of the sequences are of low quality there could be some kind of systematic problem.</p>
<p>In plant breeding data, poor per-sequence quality can indicate issues such as:</p>
<ul>
<li>Problems during library preparation.</li>
<li>Technical issues during sequencing.</li>
<li>Low-quality starting material.</li>
</ul>
</section>
<section id="per-base-sequence-content" class="level3">
<h3 class="anchored" data-anchor-id="per-base-sequence-content"><a href="">Per Base Sequence Content</a></h3>
<p><img src="figures/qc/Base_sequence_content.png" class="class img-fluid" style="width:50.0%"></p>
<p>This module shows the proportion of the bases - A, C, G, and T - at each position in the read. For random genomic DNA, these proportions should be relatively stable across the read length.</p>
<p>Some types of library will always produce biased sequence composition, normally at the start of the read. Not every deviation from equal base content is a problem. The key question is whether the pattern is expected given the experiment.</p>
<p>In plant genomes, GC content can vary naturally between regions, and this can influence this plot.</p>
</section>
<section id="per-sequence-gc-content" class="level3">
<h3 class="anchored" data-anchor-id="per-sequence-gc-content"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/5%20Per%20Sequence%20GC%20Content.html">Per Sequence GC Content</a></h3>
<p><img src="figures/qc/gc_content.png" class="class img-fluid" style="width:50.0%"></p>
<p>The GC content plot shows the overall GC distribution of reads in the sample. FastQC compares the observed distribution to a theoretical expectation, and can help identify contamination.</p>
<p>The <strong>expectation is a roughly normal distribution</strong> of GC content where the peak represents the overall GC content of the underlying genome. However, the acutal shape of the curve depends highly on your sample.</p>
<p>In plant datasets, unusual GC content does not automatically mean contamination.</p>
<p>It can reflect:</p>
<ul>
<li>Genome composition</li>
<li>Repetitive regions</li>
<li>Targeted sequencing approaches</li>
</ul>
<p>This plot becomes <strong>more informative when comparing samples to each other</strong>, rather than judging a single sample in isolation.</p>
</section>
<section id="per-base-n-content" class="level3">
<h3 class="anchored" data-anchor-id="per-base-n-content"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/6%20Per%20Base%20N%20Content.html">Per Base N Content</a></h3>
<p><img src="figures/qc/n_content.png" class="class img-fluid" style="width:50.0%"></p>
<p>If a sequencer is unable to make a base call with sufficient confidence it will call an “N” rather than a conventional base.</p>
<p>It is not unusual to see a very low proportion of Ns appearing in a sequence, especially closer to the end of a sequence.</p>
</section>
<section id="sequence-length-distribution" class="level3">
<h3 class="anchored" data-anchor-id="sequence-length-distribution"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/7%20Sequence%20Length%20Distribution.html">Sequence length distribution</a></h3>
<p><img src="figures/qc/sequence_length.png" class="class img-fluid" style="width:50.0%"></p>
<p>A graph showing the distribution of fragment sizes in the file which was analysed. Here, you can see a sample with many short, and a few very long reads.</p>
<p>Some high throughput sequencers generate sequence fragments of uniform length, but others can contain reads of varying lengths.</p>
</section>
<section id="sequence-duplication-levels" class="level3">
<h3 class="anchored" data-anchor-id="sequence-duplication-levels"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/8%20Duplicate%20Sequences.html">Sequence Duplication Levels</a></h3>
<p><img src="figures/qc/duplication_levels.png" class="class img-fluid" style="width:50.0%"></p>
<p>This plot shows how often identical sequences occur in the data. In a diverse library most sequences will occur only once in the final set, so most sequences should fall into the far left of the plot</p>
<p>High duplication can arise from:</p>
<ul>
<li>PCR amplification bias.</li>
<li>Low library complexity.</li>
<li>Very deep sequencing.</li>
</ul>
<p>In plant breeding data, duplication is not always technical. Highly conserved or repetitive regions can also produce high duplication levels.</p>
<p>Interpretation depends strongly on the type of experiment and the genome being studied.</p>
</section>
<section id="overrepresented-sequences" class="level3">
<h3 class="anchored" data-anchor-id="overrepresented-sequences"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/9%20Overrepresented%20Sequences.html">Overrepresented sequences</a></h3>
<p><img src="figures/qc/overrepresented.png" class="class img-fluid" style="width:50.0%"></p>
<p>FastQC also identifies sequences that occur much more often than expected. A <strong>normal high-throughput library will contain a diverse set of sequences</strong>, with no individual sequence making up a substantial fraction of the whole.</p>
<p>Overrepresented sequences might be biologically significant, or indicate that the library is contaminated, or simply not as diverse as expected. They can be adapter fragments, primers or highly repetitive genomic regions.</p>
<p>This module is often useful for diagnosing specific technical issues, especially when combined with adapter content and duplication plots.</p>
</section>
<section id="optional-kmer-content" class="level3">
<h3 class="anchored" data-anchor-id="optional-kmer-content"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/11%20Kmer%20Content.html">optional: Kmer content</a></h3>
<p><img src="figures/qc/kmer_profiles.png" class="class img-fluid" style="width:50.0%"></p>
<p>The k-mer content module looks for <strong>short sequence motifs</strong> that occur more often than expected in the reads.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>A <em>k-mer</em> is simply a short DNA sequence of length <em>k</em>.<br>
For example, if k = 7, a k-mer could be something like <code>ATCGTGA</code>. Fixed size kmers can also be called after their number: 3-mer, 7-mer etc.</p>
</div>
</div>
<p>FastQC scans all reads and checks whether certain k-mers are strongly enriched at specific positions along the read. In <strong>random genomic sequencing data, short sequence motifs should be more or less evenly distributed</strong>, so that is the basic assumption of FastQC.</p>
<p>This module measures the number of each 5-mer (default, but adjustable) at each position in your library and then looks for significant deviations from an even coverage at all positions. Any k-mers with positionally biased enrichment are reported. The top 6 most biased Kmer are additionally plotted to show their distribution.</p>
<p>When FastQC detects a strong enrichment of specific k-mers, this often indicates that something <strong>systematic</strong> is present in the data. Enrichment at the <strong>start or end of reads</strong> is especially common and often points to technical sequences rather than biological signal.</p>
<p>Common causes include:</p>
<ul>
<li>Adapter or primer sequences.</li>
<li>Library preparation artefacts.</li>
<li>Technical biases introduced during sequencing.</li>
</ul>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>To save resources and time, only 2% of the sample is analysed. Results are then extrapolated to the rest of the library.</p>
</div>
</div>
</section>
<section id="adapter-content" class="level3">
<h3 class="anchored" data-anchor-id="adapter-content"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/10%20Adapter%20Content.html">Adapter content</a></h3>
<p><img src="figures/qc/adapter.png" class="class img-fluid" style="width:50.0%"></p>
<p>This module will detect adapter content and plot it together with the identity of the adapter.</p>
<p>Good to know if you have adapter contamination, and which ones you might have to remove from the data.</p>
</section>
</section>
<section id="how-fastqc-fits-into-a-larger-workflow" class="level2">
<h2 class="anchored" data-anchor-id="how-fastqc-fits-into-a-larger-workflow">How FastQC fits into a larger workflow</h2>
<p>FastQC is usually used at least twice:</p>
<ol type="1">
<li>Once on raw data</li>
<li>Again after trimming or filtering</li>
</ol>
<p>Comparing FastQC reports before and after processing helps you evaluate whether your actions improved the data.</p>
<p>FastQC is therefore not a one-time checkpoint, but part of an iterative process.</p>
</section>
<section id="interactive-activity-interpreting-a-real-fastqc-report" class="level2">
<h2 class="anchored" data-anchor-id="interactive-activity-interpreting-a-real-fastqc-report">Interactive activity: interpreting a real FastQC report</h2>
<p>So, now let’s have a look at some actual FastQC reports. Pick one of the below reports, take half an hour (with a partner if you have one at hand) and try to understand what is going on. Keep in mind what sort of data you are working with.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Remember, most often, there is no single “correct” answer. What matters is whether your interpretation is <strong>biologically and technically justified.</strong></p>
</div>
</div>
<p>Here are some guiding questions for you:</p>
<ol type="1">
<li>Looking at the summary table, which modules are marked as warnings or failures? Which ones would you look at first, and why?</li>
<li>How does read quality change along the length of the reads? Is the quality along the sample problematic, or acceptable?</li>
<li>Is there evidence of adapter contamination or overrepresented sequences?</li>
<li>Based on this report, would you: i) Use the data as is? ii) Apply trimming or filtering? iii) Be concerned about the sample and investigate further?</li>
<li>Are there any FastQC warnings you would ignore for this dataset? Why are they acceptable in this context?</li>
<li>If you had three samples and only this one looked different, what would you want to check next?</li>
</ol>
<p><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/RNA-Seq_fastqc.html#M10">Report 1: RNA-sequencing data</a> short read data - remember that RNA sequencing does not look at the whole genome.</p>
<p><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/pacbio_srr075104_fastqc.html#M7">Report 2: PacBio Data</a>, long read data - see how different this type of data looks like in the FastQC report.</p>
<p><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html#M0">Report 3: bad sample data</a> short read data - can you figure out what happened to this sample?</p>
<p><a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#assess-quality-with-fastqc---short--long-reads">Here</a> is a very nice and thorough tutorial on assessing quality of short and long read sequencing data. Great to have a look at the difference the sequencing technology can make!</p>
</section>
<section id="after-quality-control-what-comes-next" class="level2">
<h2 class="anchored" data-anchor-id="after-quality-control-what-comes-next">After quality control: what comes next?</h2>
<p>After working through real FastQC reports, one important point should be clear: Quality control is not about deciding whether data is “good” or “bad”. It is about deciding <strong>what to do next</strong>.</p>
<p>Sequencing data almost always contains imperfections. The goal of QC is to understand whether these imperfections matter for your biological question, and how they should be handled.</p>
</section>
<section id="acting-on-qc-results" class="level2">
<h2 class="anchored" data-anchor-id="acting-on-qc-results">Acting on QC results</h2>
<p>Once you have interpreted a FastQC report, there are usually three possible outcomes.</p>
<ol type="1">
<li><p>In some cases, the data can be used <strong>as it is</strong>. Minor warnings or expected patterns do not necessarily require any action, especially if they are unlikely to affect downstream analyses.</p></li>
<li><p>In other cases, <strong>simple processing steps</strong> are appropriate. This often includes trimming adapters or low-quality read ends. After such processing, QC should be repeated to confirm that the data has improved.</p></li>
<li><p>In a small number of cases, QC reveals <strong>serious problems</strong>. These may require investigating sample preparation, resequencing, or excluding a sample from further analysis.</p></li>
</ol>
<p>At the beginning it might be a lot to go through the reports and judge the samples, but with time and practice it will come more natural to you.</p>
</section>
<section id="qc-as-part-of-a-reproducible-workflow" class="level2">
<h2 class="anchored" data-anchor-id="qc-as-part-of-a-reproducible-workflow">QC as part of a reproducible workflow</h2>
<p>Quality control should not be an informal or invisible step.</p>
<p>Good practice includes <strong>running QC in a consistent way across all samples</strong>, <strong>recording which issues were observed</strong> and <strong>documenting which actions were taken, and why</strong>.</p>
<p>This is especially important in plant breeding projects, where results may influence long-term decisions.</p>
<p>Clear QC reporting makes analyses more transparent and easier to reproduce.</p>
</section>
<section id="bonus-multiqc" class="level2">
<h2 class="anchored" data-anchor-id="bonus-multiqc">Bonus: MultiQC</h2>
<p>So far, we have focused on <strong>one FastQC report at a time</strong>. In real projects, especially in plant breeding, we usually work with <strong>many samples</strong>.</p>
<p>When dozens or hundreds of samples are sequenced, it becomes tedious to interpret QC results sample by sample. This is where <strong>MultiQC</strong> is useful because is collects reports from many samples and condenses them into a single report.</p>
<p>That way you can compare samples side by side, spot outliers quickly, and identify systematic problems across samples.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/amrei-bp\.github\.io\/emplant_bioinformatics_course\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>